pkg/detector/library/advisory_test.go

```diff
@@ -66,12 +68,13 @@ func TestAdvisory_DetectVulnerabilities(t *testing.T) {
 			},
 		},
 		{
-			name:     "no vulnerable versions in the advisory",
-			fixtures: []string{"testdata/fixtures/ruby.yaml"},
-			fields:   fields{lang: vulnerability.Ruby},
+			name:      "no vulnerable versions in the advisory",
+			fixtures:  []string{"testdata/fixtures/ruby.yaml"},
+			ecosystem: vulnerability.RubyGems,
+			comparer:  bundler.RubyGemsComparer{},
 			args: args{
 				pkgName: "activesupport",
-				pkgVer:  semver.MustParse("4.1.1"),
+				pkgVer:  "4.1.1",
```

Comment:
For my own understanding: `~> X` in ruby land means `>= X AND <X.1.0`?

Reply:
More complex 😄  You can see the following examples. It depends on how many dot-separated numbers are specified.
https://github.com/aquasecurity/go-gem-version/blob/main/README.md#pessimistic-operator

In this case, `~> 4.1.11` means `>= 4.1.11, < 4.2` and `4.1.1` satisfies the constraint.